CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -770,21 +769,43 @@ public static function recurringContribution(&$form) {
 
   }
 
+  /**
+   * Get the metadata for fields to be included on the search form.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public static function getContributionRecurSearchFieldMetadata() {
+    $fields = [
+      'contribution_recur_start_date' => ['title' => ts('Recurring Contribution Start Date')],
```

Comment:
@seamuslee001 @colemanw @mattwire @totten @mlutfy  this highlights the issue I got stuck on when I started this piece of work earlier (& adding cancel_reason) - which is that we have 2 title needs

1) Context specific - e.g on the Recurring Contribution form saying End Date makes sense 
2) Context unknown - e.g on advanced search form or search builder or views there might be more than one 'end date' and adding it to the title makes sense. (The accordians give some context but one you have searched you just have the qill)

I'm sure @mlutfy would not advise we do what is currently being erratically done in query object for qill & concat the entity with the title - I think probably we should instead add a new field to the schema for 'context_lite_title' - I'm not sure the best name though

As a side issue we might also think about the title needs for id fields which are used both as ID fields (views, search builder) and as entity reference fields where to all intents & purposes they refer to 'the thing' - ie. filling in the 'financial_type_id' for a contribution. Once again I feel like we need 2 different titles here. What I have been doing is in fact deriving the title - e.g if the link is to civicrm_financial_type.name then getting the title for that field or for option group references getting the option group label - but changes I made in this direction were fighting @colemanw's changes to better support entity reference fields & other stuff

Reply:
Sounds like we need something like a `unique_title` to go with (my favorite schema field) `unique_name`.

Reply:
@seamuslee001 I've opened https://github.com/civicrm/civicrm-core/pull/14737 to add unique_title just to make @colemanw happy :-)